illini-robomaster/iRM_Embedded_2023
# Add minipc protocol 2.0

Update Minipc - Stm32 uart protocol.
### Major feature: 
- support direct float number transmit
- support varying packet length

### Minor changes:
- add pingpong test example
- add vanilla sending example
- add stresstest example (max rate at ~700Hz)
- add motor example

Note that the communication should always be single-sided. A response should never **directly** depend on any data in previous packets. Example: send back `rel_pitch += 1` is an undefined behavior. (Yes, this is the pingpong test.)

### Protocol details:
See document: `docs/comm_protocol.md` in illini-robomaster/iRM_Vision_2023#16

### To test this: 
Embedded repo `example/minipc/StressTestTypeC.cc` with Vision repo `Communication/communicator.py`
and
Embedded repo `example/minipc/PingpongTest.cc` with Vision repo `Communication/communicator.py`
For testing details and config setup see code comments.

Embedded PR: #66
Vision PR: illini-robomaster/iRM_Vision_2023#16

### Todo: 
- [ ] change autoaim thread in vehicle/steering
- [x] add typeA example
- [ ] change autoaim example
